src/app/reporting/Engine.cpp

```diff
@@ -289,8 +290,11 @@ CHIP_ERROR Engine::BuildSingleReportDataEventReports(ReportDataMessage::Builder
     }
 
     {
-        EventReportIBs::Builder & eventReportIBs = aReportDataBuilder.CreateEventReports();
+        const uint32_t kReservedSizeEndOfReportIBs = 1;
```

Comment:
So we're reserving 1 byte?  Might be worth clearly documenting why one byte (for the TLV end-of-struct marker, I am assuming?).

Reply:
Added a comment for it